Remove all usages of MailPoet\Tasks\Sending::getScheduledQueues() [MAILPOET-4367]

lysyjan opened the pull request:
## Description
This pull request starts with https://github.com/mailpoet/mailpoet/commit/cfc359b56fa6b22d5efab158793dc3811b984b57 because I wanted to avoid conflicts.

It should be merged after https://github.com/mailpoet/mailpoet/pull/4305 and rebased on the trunk before.

## QA notes
Make sure scheduled newsletter sending works as expected

## Linked tickets
[MAILPOET-4367]

[MAILPOET-4367]: https://mailpoet.atlassian.net/browse/MAILPOET-4367?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

lysyjan committed:
Use currentTimme from WP

[MAILPOET-4367]

lysyjan committed:
Add method for finding scheduled tasks

[MAILPOET-4367]

lysyjan committed:
Remove usage of Sending::getScheduledQueues

[MAILPOET-4367]

lysyjan committed:
Remove the Sending::getScheduledQueues() method

[MAILPOET-4367]

lysyjan committed:
Use ScheduledTaskEntity instead of Sending

[MAILPOET-4367]

lysyjan committed:
Add integration test for findScheduledSendingTasks

[MAILPOET-4367]

JanJakes commented:
@rodrigoprimo Could you pls have a look at this? 🙏 I didn't manage to look at this before my AFK. If that doesn't suit you, feel free to assign it to anybody else. Thanks!

rodrigoprimo commented:
@JanJakes, sure, I can review this PR. Enjoy your time off.

@lysyjan, would it be ok if I wait until #4305 is merged and this PR rebased to review it? #4305 is already on QA so it is likely to be merged soon.

rodrigoprimo commented on mailpoet/tests/integration/Cron/Workers/SchedulerTest.php:
```diff
@@ -760,6 +761,7 @@ public function testItProcessesWelcomeNewsletters() {
     $scheduler = Stub::make(Scheduler::class, [
```
Maybe instead of adding the new dependency of the Scheduler class, we can use \MailPoetTest::getServiceWithOverrides() to avoid having to update this test again if a new dependency is added to the same class?

lysyjan replied:
This was implemented in a different PR. Here is the commit: https://github.com/mailpoet/mailpoet/commit/bfc2a7d2afc0e148a3440b80f53d770fa5a10d28

rodrigoprimo replied:
:+1:

rodrigoprimo commented on mailpoet/lib/Newsletter/Sending/ScheduledTasksRepository.php:
```diff
@@ -156,6 +156,27 @@ public function findRunningSendingTasks(?int $limit = null): array {
+  /**
+   * @return ScheduledTaskEntity[]
+   */
+  public function findScheduledSendingTasks(int $limit): array {
```
`Sending::getScheduledQueues()` has a default value for `$amount` (`Sending::RESULT_BATCH_SIZE`) and this new method doesn't. I don't it that is necessary, just checking if that was a conscious decision and curious about the reasoning behind it.

lysyjan replied:
I didn't want to use a constant from a different class as a default value in the repository. In the end, I decided to unify the limit parameter with a similar method `findRunningSendingTasks` from this repository. This change was added as a new commit.

rodrigoprimo replied:
:+1:

rodrigoprimo reviewed (dismissed):
Great work, @lysyjan. I left two comments that are not blockers. Feel free to move this PR directly to QA.

lysyjan committed:
Make limit parameter optional

Unification function parameters with a similar method from the repository.
[MAILPOET-4367]

veljkho commented:
Hi @lysyjan 
There are 3 conflicting files found.

lysyjan commented:
@veljkho Done

Aschepikov reviewed (approved):
This is verified, good job.
